File: app/static/js/findface.js
$(function(){
    $('#face').on('change', function(){
        img = getObjectURL(this.files[0]);
        $('#preview').attr('src', img);
    });

    $('#upload').on('click', function(){
        formdata = new FormData();
        img = $('#face')[0].files[0];
        console.log(img);
        formdata.append('photo', img);
        formdata.append('dirname', 'face');
        $.toptips('正在上传', 'success');
        $.ajax({
            url: BASE_URL + '/common/upload_img',
            type: 'post',
            contentType: false,
            processData: false,
            data: formdata,
            success: function(resp){
                if(resp.success){
                    $.toptips('上传成功', 'success');
                }else{
                    $.toptips('上传失败');
                }
            },
            error: function(){
                $.toptips('服务器错误');
            }
        });
    });
    $('#detect').on('click', function(){
        api_url = 'http://jswj888.f3322.net:8889';
        formdata = new FormData();
        img = $('#face')[0].files[0];
        console.log(img);
        formdata.append('photo', img);
        $.ajax({
            url: api_url + '/n-tech/v0/detect',
            type: 'post',
            headers: {'Authorization': 'Token tGd3-nXYt'},
            contentType: false,
            processData: false,
            data: formdata,
            success: function(resp){
                console.log(resp);
            },
            error: function(){
                console.log('error');
            }
        }); 
    });
});

File: app/service/findface.py
#-*- coding:utf-8 -*-
import logging
import requests


api_url = 'http://jswj888.f3322.net:8889'
def detect(path):
    headers = {
        'Authorization': 'Token tGd3-nXYt',
    }
    files = {
        'photo': ('face', open(path, 'rb'), 'image/jpeg'),
    }
    r = requests.post(api_url + '/n-tech/v0/detect', headers=headers, files=files)
    print('REQUEST HEADERS:', r.request.headers)
    print('RESPONSE', r)
    print('CONTENT:', r.json())


def identify(path):
    headers = {
        'Authorization': 'Token tGd3-nXYt',
    }
    files = {
        'photo': ('face', open(path, 'rb'), 'image/jpeg'),
    }
    r = requests.post(api_url + '/n-tech/v0/identify', headers=headers, files=files)
    print('REQUEST HEADERS:', r.request.headers)
    print('RESPONSE', r)
    print('CONTENT:', r.json())


def verify(path1, path2):
    headers = {
        'Authorization': 'Token tGd3-nXYt',
    }
    files = { 
        'photo1': open(path1, 'rb'),
        'photo2': open(path2, 'rb'),
    }
    r = requests.post(api_url + '/n-tech/v0/verify', headers=headers, files=files)
    print('REQUEST HEADERS:', r.request.headers)
    print('RESPONSE', r)
    print('CONTENT:', r.json())

